wamr-compiler/CMakeLists.txt

```diff
@@ -198,32 +197,82 @@ include_directories (${SHARED_DIR}/include
 
 enable_language (ASM)
 
-if (NOT MINGW AND NOT MSVC)
-    set(WAMR_BUILD_LIBC_WASI 1)
+if (NOT DEFINED WAMR_BUILD_LIBC_WASI)
+  if (NOT MINGW AND NOT MSVC)
+      set(WAMR_BUILD_LIBC_WASI 1)
+  else()
+      set(WAMR_BUILD_LIBC_UVWASI 1)
+  endif()
+endif()
+
+if (NOT DEFINED WAMR_BUILD_LIBC_BUILTIN)
+  set(WAMR_BUILD_LIBC_BUILTIN 1)
+endif()
+
+if (NOT DEFINED WAMR_BUILD_LIB_PTHREAD)
+  set(WAMR_BUILD_LIB_PTHREAD 1)
+endif()
+
+if (NOT DEFINED WAMR_BUILD_LIB_WASI_THREADS)
+  set(WAMR_BUILD_LIB_WASI_THREADS 1)
+endif()
+
+if(WAMR_BUILD_LIBC_WASI EQUAL 1)
+  message ("-- Libc WASI enabled")
+else()
+  message ("-- Libc WASI disable")
+endif()
+
+if(WAMR_BUILD_LIBC_BUILTIN EQUAL 1)
+  message ("-- Libc builtin enabled")
 else()
-    set(WAMR_BUILD_LIBC_UVWASI 1)
+  message ("-- Libc builtin disable")
 endif()
 
+if (WAMR_BUILD_LIB_PTHREAD EQUAL 1)
+  message ("-- Lib pthread enabled")
+else()
+  message ("-- Lib pthread disable")
+endif ()
+
+if (WAMR_BUILD_LIB_WASI_THREADS EQUAL 1)
+  message ("-- Lib wasi-threads enabled")
+else()
+  message ("-- Lib wasi-threads disable")
+endif ()
+
 include (${SHARED_DIR}/platform/${WAMR_BUILD_PLATFORM}/shared_platform.cmake)
 include (${SHARED_DIR}/mem-alloc/mem_alloc.cmake)
 include (${SHARED_DIR}/utils/shared_utils.cmake)
 include (${SHARED_DIR}/utils/uncommon/shared_uncommon.cmake)
 include (${IWASM_DIR}/libraries/thread-mgr/thread_mgr.cmake)
-include (${IWASM_DIR}/libraries/libc-builtin/libc_builtin.cmake)
-if (NOT MINGW)
-  if (NOT MSVC)
-    include (${IWASM_DIR}/libraries/libc-wasi/libc_wasi.cmake)
-  else()
-    include (${IWASM_DIR}/libraries/libc-uvwasi/libc_uvwasi.cmake)
-  endif()
-endif()
-include (${IWASM_DIR}/libraries/lib-pthread/lib_pthread.cmake)
-include (${IWASM_DIR}/libraries/lib-wasi-threads/lib_wasi_threads.cmake)
 include (${IWASM_DIR}/common/iwasm_common.cmake)
 include (${IWASM_DIR}/interpreter/iwasm_interp.cmake)
 include (${IWASM_DIR}/aot/iwasm_aot.cmake)
 include (${IWASM_DIR}/compilation/iwasm_compl.cmake)
 
+if (WAMR_BUILD_LIBC_BUILTIN EQUAL 1)
+  include (${IWASM_DIR}/libraries/libc-builtin/libc_builtin.cmake)
+endif()
+
+if (WAMR_BUILD_LIBC_WASI EQUAL 1)
+  if (NOT MINGW)
+    if (NOT MSVC)
+      include (${IWASM_DIR}/libraries/libc-wasi/libc_wasi.cmake)
+    else()
+      include (${IWASM_DIR}/libraries/libc-uvwasi/libc_uvwasi.cmake)
```

Comment:
i feel it's better to use WAMR_BUILD_LIBC_UVWASI to control libc_uvwasi to align better with config_common.cmake.

Reply:
So, it should be like below:
```cmake
if (NOT MINGW AND NOT MSVC)
  if (NOT DEFINED WAMR_BUILD_LIBC_WASI)
    set(WAMR_BUILD_LIBC_WASI 1)
  endif()
else ()
  if (NOT DEFINED WAMR_BUILD_LIBC_UVWASI)
    set(WAMR_BUILD_LIBC_UVWASI 1)
  endif()
endif()
```
yes, seems it is better.